tests/wast.rs

```diff
@@ -10,15 +11,188 @@ use wasmtime::{
 use wasmtime_environ::WASM_PAGE_SIZE;
 use wasmtime_wast::{SpectestConfig, WastContext};
 
-include!(concat!(env!("OUT_DIR"), "/wast_testsuite_tests.rs"));
+fn main() {
+    env_logger::init();
+
+    let mut trials = Vec::new();
+    add_tests(&mut trials, "tests/spec_testsuite".as_ref());
+    add_tests(&mut trials, "tests/misc_testsuite".as_ref());
+
+    // There's a lot of tests so print only a `.` to keep the output a
+    // bit more terse by default.
+    let mut args = Arguments::from_args();
+    if args.format.is_none() {
+        args.format = Some(FormatSetting::Terse);
+    }
+    libtest_mimic::run(&args, trials).exit()
+}
+
+fn add_tests(trials: &mut Vec<Trial>, path: &Path) {
+    for entry in path.read_dir().unwrap() {
+        let entry = entry.unwrap();
+        let path = entry.path();
+        if entry.file_type().unwrap().is_dir() {
+            add_tests(trials, &path);
+            continue;
+        }
+
+        if path.extension().and_then(|s| s.to_str()) != Some("wast") {
+            continue;
+        }
+
+        for strategy in [Strategy::Cranelift, Strategy::Winch] {
+            for pooling in [true, false] {
+                let trial = Trial::test(
+                    format!(
+                        "{strategy:?}/{}{}",
+                        if pooling { "pooling/" } else { "" },
+                        path.to_str().unwrap()
+                    ),
+                    {
+                        let path = path.clone();
+                        move || {
+                            run_wast(&path, strategy, pooling).map_err(|e| format!("{e:?}").into())
+                        }
+                    },
+                );
+                trials.push(trial.with_ignored_flag(ignore(&path, strategy)));
+            }
+        }
+    }
+}
+
+fn ignore(test: &Path, strategy: Strategy) -> bool {
+    if cfg!(miri) {
+        return true;
+    }
```

Comment:
If all tests are ignored on miri, can we just avoid constructing any trials in the first place?